src/commands/install/utils/map_url_to_package_name.py

```diff
@@ -0,0 +1,16 @@
+import re
+from typing import Optional
+
+
+def map_url_to_package_name(url: str) -> Optional[str]:
```

Comment:
i think we should throw whenever we encounter a mapping error, this will prevent us from having `None`s in the flow later which would be hard to catch